RapidWebView/src/main/java/in/workindia/rapidwebview/utils/RapidBrodCastActionUtility.kt

```diff
@@ -0,0 +1,183 @@
+     * event to generate and dispatch based on the provided intent's status.
+     * event: "rapid-web-view-upload-listener" for upload success or failure
+     * event: "rapid-web-view-upload-listener" for permissions success or failure
+     *
+     * @see [UploadService]
+     * @see [RapidWebViewJSInterface.broadcastReceiverForNativeCallbacks]
+     * @see [permissionSuccessHelper]
+     * @param intent The intent containing callback action data.
+     * @param webView The WebView to which the JavaScript will be dispatched.
+     * @param context The context in which the function is executed. Can be null.
+     */
+    fun handleNativeCallbackAction(intent: Intent, webView: WebView, context: Context?) {
+        val permissionStatus = intent.getStringExtra(BroadcastConstants.PERMISSION)
+        val uploadStatus = intent.getStringExtra(BroadcastConstants.UPLOAD)
+        val uploadUrl = intent.getStringExtra(BroadcastConstants.UPLOADED_URL) ?: ""
+        val fileName = intent.getStringExtra(BroadcastConstants.UPLOADED_FILE_NAME) ?: ""
+        val permissionList =
+            intent.getStringArrayExtra(BroadcastConstants.PERMISSION_LIST) ?: emptyArray()
+
+        val javaScript = when {
+            uploadStatus.equals(BroadcastConstants.SUCCESS, ignoreCase = true) -> {
+                generateJavaScriptEvent(
+                    "rapid-web-view-upload-listener", mapOf(
+                        "status" to "success",
+                        "uploadUrl" to uploadUrl,
+                        "uploadFileName" to fileName
+                    )
+                )
+            }
+
+            uploadStatus.equals(BroadcastConstants.FAILURE, ignoreCase = true) -> {
+                generateJavaScriptEvent(
+                    "rapid-web-view-upload-listener", mapOf(
+                        "status" to "failure"
+                    )
+                )
+            }
+
+            permissionStatus.equals(BroadcastConstants.SUCCESS, ignoreCase = true) -> {
+                permissionSuccessHelper(permissionList, intent, context)
+            }
+
+            permissionStatus.equals(BroadcastConstants.FAILURE, ignoreCase = true) -> {
+                generateJavaScriptEvent(
+                    "rapid-web-view-permission-listener", mapOf(
+                        "status" to "failure"
+                    )
+                )
+            }
+
+            else -> ""
+        }
+
+        if (javaScript.isNotBlank()) {
+            dispatchJavaScript(javaScript, webView)
+        }
+    }
+
+
+    /**
+     * Handles the action to be taken when a download is complete.
+     * Displays a toast message and opens the downloaded file.
+     *
+     * @param context The context received from BrodCastReceiver.
+     * @param intent The intent received from BrodCastReceiver.
+     */
+    fun handleDownloadCompleteAction(context: Context?, intent: Intent) {
+        Toast.makeText(context, "Download Complete", Toast.LENGTH_SHORT).show()
```

Comment:
Instead of showing Toast message, send a `generateJavaScriptEvent` so the developers can handle it as per their taste.